### PR TITLE
Correct handling of floats test sensor list

### PR DESCRIPTION
**Pull request by tockards**
This PR corrects the handling of floats in the`sensor_tuple` helper function used in `test_sensor_list`.

- This adds a new float sensor with a zero to the right of the decimal point 
- Tests where the sensor is tested correctly inside `test_sensor_list`

### CODE REVIEW

File: katcp/test/test_server.py [hunk 1]
```diff
@@ -661,9 +669,7 @@ def tst():
             (r"#log error", r"root error-msg"),
             (r"#log fatal", r"root fatal-msg"),
         ]
-        self._assert_msgs_like(get_msgs(min_number=len(expected_msgs)),
-                               expected_msgs)
-
+        self._assert_msgs_like(get_msgs(min_number=len(expected_msgs)), expected_msgs)
     def test_standard_requests_with_ids(self):
```
Comment (ajoubertza) [inline katcp/test/test_server.py:673]: Missing blank line before the method?

**Review by ajoubertza — APPROVED** [review]
Looks good, except for the missing blank line.

**Comment by tockards** [comment]
@ajoubertza Thanks,  this creeps in when I do my black format on a selected part of code and commit in a rush.

**Comment by tockards** [comment]
Need to add the linting option to tox so that  a developer can do it locally.

File: katcp/testutils.py [hunk 2]
```diff
@@ -1045,6 +1045,10 @@ def setup_sensors(self):
             Sensor.FLOAT, "a.float", "A Float.", "",
             [-123.4, 123.4],
             timestamp=12345, status=Sensor.NOMINAL, value=12))
+        self.add_sensor(DeviceTestSensor(
```
Comment (mmphego) [inline katcp/testutils.py:1048]: My OCD got triggered :rofl: 

```suggestion
        self.add_sensor(
            DeviceTestSensor(
                Sensor.FLOAT, 
                "a.floatwithzero", 
                "A Float with zero.", 
                "",
                [-123.0, 123.0],
                timestamp=12345,
                status=Sensor.NOMINAL,
                value=12))
```

**Review by mmphego — APPROVED** [review]
Looks good to me.

**Comment by bmerry** [comment]
I've started using [pre-commit](https://pre-commit.com/) to define pre-commit hooks - it's a very nice Python wrapper around Git's hook system and it's saved me from making a lot of commits that would have failed flake8 or mypy.

File: katcp/testutils.py (hunk 2 above)
Comment (tockards) [reply katcp/testutils.py:1048]: @mmphego Do you want me to run black on all the code?

**Comment by tockards** [comment]
thanks @bmerry . Going to give it a try.

File: katcp/testutils.py (hunk 2 above)
Comment (mmphego) [reply katcp/testutils.py:1048]: Would be nice to run black before committing IMHO.

Comment (tockards) [reply katcp/testutils.py:1048]: thanks but from my experience black conflicts with our linter. see http://ci.camlab.kat.ac.za/blue/organizations/jenkins/katcp-multibranch/detail/user%2Ftockards%2Fblack-test-server/1/pipeline

Unless there are settings and things I need to suppress in black.

Comment (mmphego) [reply katcp/testutils.py:1048]: You could run it as: `black -t py27 -t py36 -l 90 <file>`

**Pull request merged**